accepted/00000-rbac.md

```diff
@@ -0,0 +1,271 @@
+# RFC: Role-Based Access Control
+
+- Feature Name: Role-Based Access Control (RBAC)
+- Start Date: 2024-11-12
+- Author: @cbbayburt
+
+# Summary
+[summary]: #summary
+
+This RFC proposes implementing role-based access control (RBAC) in Uyuni to meet users' needs for fine-grained access control and configuration.
+The solution is extensible, integrates with existing features of the codebase, and is independent of the ongoing efforts to migrate from the Struts framework.
+
+
+# Motivation
+[motivation]: #motivation
+
+Currently, access control rules are defined by static role information and controlled individually within each action.
+By default, each new feature must define its own access rules; if no rules are explicitly defined, the feature is available to all authenticated users.
```

Comment:
Is this the default we want to go with? I'm not entirely sure which makes more sense, available to all by default or available to none by default. My gut feel is that available to none by default would be more sensible, but I think regardless of which one we go with, we should have a good reason for picking one over the other.

Reply:
No that's the current default. Proposed way is the opposite, mentioned down below:

> By default, any requested URI is denied access unless a match is found.

It's especially important because it'll also act as failsafe when a dev forgets to map an endpoint, they won't be able to access it when testing.

Reply:
Ah okay, yes, I agree with that logic. I think I misunderstood which sentence talks about the old bits and which about new.